### PR TITLE
Add Server pdf format options

### LINKED ISSUE #8292: Server pdf format options (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#52799
Author: @mhugent
QGIS version: 3.32

**Server pdf format options**

### PR Description:
In QGIS Server, there is the FORMAT_OPTIONS parameter to pass format options for DXF format to WMS GetPrint. This PR adds support for format options for PDF format in WMS GetPrint

### Commits tagged with [need-docs] or [FEATURE]
"[FEATURE] WMS server: possibility to set pdf export options (as FORMAT_OPTIONS) in the GetPrint request"

### PR DESCRIPTION
**Pull request by DelazJ**
This is an attempt to fix #8292 
It assumes I understand what it is about because of the various points I already raised at https://github.com/qgis/QGIS/pull/52799#issuecomment-1614265126. I could have put it in draft mode but considering that I can't do anything more, better have it reviewed by those who actually know, and fill the `???` gaps in the PR.

@mhugent ?

### CODE REVIEW

File: docs/server_manual/services/wms.rst [hunk 1]
```diff
@@ -1733,6 +1736,50 @@ This parameter specifies the format of map image. Available values are:
...
+
+FORMAT_OPTIONS
+^^^^^^^^^^^^^^
+
+This parameter can be used to specify options for the selected format.
+Only for ``FORMAT=application/pdf`` in GetPrint requests.
+Takes a list of key:value pairs separated by semicolon:
+
+.. keys picked at https://github.com/qgis/QGIS/pull/52799/files#diff-f2b6ac14970527d38f8137233a258cbde7733fac3b6578022a45ef4d33374669
+ but descriptions and their default values need deep review
```
Comment (DelazJ) [inline docs/server_manual/services/wms.rst:1749]: indeed!!!

**Review by DelazJ — COMMENTED** [review]
Raising points I'm in doubt with into comments. Hopefully they get replies...

**Comment by DelazJ** [comment]
I don't think this PR should have been closed, so let's give it another chance.
@mhugent I'd appreciate if you can help fill in the gaps in this pull request. I'm not a QGIS Server user. Thanks

File: docs/server_manual/services/wms.rst [hunk 2]
```diff
@@ -1733,6 +1736,50 @@ This parameter specifies the format of map image. Available values are:
...
+* FORCE_VECTOR_OUTPUT: whether pdf should be exported as vector. Default: false.
+* APPEND_GEOREFERENCE: whether georeference info shall be added to the pdf. Default: true.
+* EXPORT_METADATA: whether metadata shall be added to the pdf. Default: true.
+* TEXT_RENDER_FORMAT: sets the text render format for pdf export.
+  It can be ``AlwaysOutlines`` (default) or ``AlwaysText``.
+* SIMPLIFY_GEOMETRY: whether features geometries shall be simplified. Default: true.
+* WRITE_GEO_PDF: whether a GeoPDF shall be exported. Default: false.
+* USE_ISO_32000_EXTENSION_FORMAT_GEOREFERENCING: whether Iso32000 georeferencing shall be used. Default: false.
+* USE_OGC_BEST_PRACTICE_FORMAT_GEOREFERENCING: whether OGC best practice georeferencing shall be used. Default: false.
+* INCLUDE_GEO_PDF_FEATURES: .... ???
```
[the remark this answers is not in the document]
Comment (mhugent) [reply docs/server_manual/services/wms.rst:1761]: INCLUDE_GEO_PDF_FEATURES is a member of the PdfFormatOption enum, but it is not used in the code (and should therefore not appear in the documentation)

File: docs/server_manual/services/wms.rst [hunk 3]
```diff
@@ -1733,6 +1736,50 @@ This parameter specifies the format of map image. Available values are:
...
+
+  http://localhost/qgisserver?
+  SERVICE=WMS
+  &VERSION=1.3.0
+  &REQUEST=GetPrint
+  &MAP=/home/qgis/projects/world.qgs
+  &CRS=EPSG:4326
+  &FORMAT=pdf
+  &TEMPLATE=Layout%201
+  &FORMAT_OPTIONS=FORCE_VECTOR_OUTPUT:TRUE;TEXT_RENDER_FORMAT:AlwaysOutlines;PREDEFINED_MAP_SCALES:250
```
[the remark this answers is not in the document]
Comment (mhugent) [reply docs/server_manual/services/wms.rst:1780]: Syntax looks correct to me

**Pull request merged**